src/playback-watcher.js

```diff
@@ -115,6 +121,31 @@ export default class PlaybackWatcher {
   }
 
   /**
+   * Fixes situations where there's a bad seek
+   *
+   * @return {Boolean} whether an action was taken to fix the seek
+   * @private
+   */
+  fixBadSeeks_() {
+    let seekable = this.seekable();
+    let currentTime = this.tech_.currentTime();
+
+    if (this.tech_.seeking() &&
```

Comment:
I don't think its bad to have this for VOD. Seekable doesn't change in VOD anyway, so if someone tries seeking outside of the duration of the video, this would just adjust it to 0 or end of video depending on which direction they overseeked